### PR TITLE
config: add option to enable placement rules

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### What problem does this PR solve?
Add an option to turn on/off placement rules feature.

### What is changed and how it works?
Add configuration and make sure it is always disabled before the full feature get merged.

### CODE REVIEW

File: server/config/config.go [hunk 1]
```diff
@@ -744,15 +744,19 @@ type ReplicationConfig struct {
 	LocationLabels typeutil.StringSlice `toml:"location-labels,omitempty" json:"location-labels"`
 	// StrictlyMatchLabel strictly checks if the label of TiKV is matched with LocationLabels.
 	StrictlyMatchLabel bool `toml:"strictly-match-label,omitempty" json:"strictly-match-label,string"`
+
+	// When PlacementRules feature is enabled. MaxReplicas and LocationLabels are not uesd any more.
+	EnablePlacementRules bool // Keep it false before full feature get merged. `toml:"enable-placement-rules" json:"enable-placement-rules,string"`
```
Comment (lhy1024) [inline server/config/config.go]: we also need add it to `toml`?

File: server/config/option.go [hunk 2]
```diff
@@ -132,6 +132,11 @@ func (o *ScheduleOption) GetLocationLabels() []string {
 	return o.replication.GetLocationLabels()
 }
 
+// GetEnablePlacementRules returns if the placment rules is enabled.
```
Comment (lhy1024) [inline server/config/option.go]:
```suggestion
// GetEnablePlacementRules returns if the placement rules is enabled.
```

File: pkg/mock/mockoption/mockoption.go [hunk 3]
```diff
@@ -183,6 +186,11 @@ func (mso *ScheduleOptions) GetStrictlyMatchLabel() bool {
 	return mso.StrictlyMatchLabel
 }
 
+// GetEnablePlacementRules mocks method
+func (mso *ScheduleOptions) GetEnablePlacementRules() bool {
```
Comment (lhy1024) [inline pkg/mock/mockoption/mockoption.go]: would you like replace it with `isPlacementRulesEnabled`？

File: server/config/config.go (hunk 1 above)
Comment (disksing) [reply server/config/config.go]: Comment it on purpose so neither config file nor pd-ctl can enable it.

File: pkg/mock/mockoption/mockoption.go (hunk 3 above)
Comment (disksing) [reply pkg/mock/mockoption/mockoption.go]: Good point

File: server/config/config.go (hunk 1 above)
Comment (lhy1024) [reply server/config/config.go]: I think we can add a comment in toml to avoid forgetting it.

**Pull request merged**